jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/JibPlugin.java

```diff
@@ -162,68 +165,49 @@ public void apply(Project project) {
             SyncMapTask.class,
             task -> task.setJibExtension(jibExtension));
 
+    Set<TaskProvider<?>> jibTaskProviders =
+        ImmutableSet.of(buildImageTask, buildDockerTask, buildTarTask, syncMapTask);
+
     // A check to catch older versions of Jib.  This can be removed once we are certain people
     // are using Jib 1.3.1 or later.
     tasks.register(SKAFFOLD_CHECK_REQUIRED_VERSION_TASK_NAME, CheckJibVersionTask.class);
 
     project.afterEvaluate(
         projectAfterEvaluation -> {
-          try {
-            TaskProvider<Task> warTask = TaskCommon.getWarTaskProvider(projectAfterEvaluation);
-            TaskProvider<Task> bootWarTask =
-                TaskCommon.getBootWarTaskProvider(projectAfterEvaluation);
-            List<TaskProvider<?>> dependsOnTask = new ArrayList<>();
-            if (warTask != null || bootWarTask != null) {
-              // Have all tasks depend on the 'war' and/or 'bootWar' task.
-              if (warTask != null) {
-                dependsOnTask.add(warTask);
-              }
-              if (bootWarTask != null) {
-                dependsOnTask.add(bootWarTask);
-              }
-            } else if ("packaged".equals(jibExtension.getContainerizingMode())) {
-              // Have all tasks depend on the 'jar' task.
-              TaskProvider<Task> jarTask = projectAfterEvaluation.getTasks().named("jar");
-              dependsOnTask.add(jarTask);
-
-              if (projectAfterEvaluation.getPlugins().hasPlugin("org.springframework.boot")) {
-                Jar jar = (Jar) jarTask.get();
-                jar.setEnabled(true);
-                jar.getArchiveClassifier().set("original");
-              }
-            } else {
-              // Have all tasks depend on the 'classes' task.
-              dependsOnTask.add(projectAfterEvaluation.getTasks().named("classes"));
+          TaskProvider<Task> warTask = TaskCommon.getWarTaskProvider(projectAfterEvaluation);
+          TaskProvider<Task> bootWarTask =
+              TaskCommon.getBootWarTaskProvider(projectAfterEvaluation);
+          TaskProvider<Task> jibDependenciesTask = tasks.register("jibDependencies");
```

Comment:
Can you use a const as above https://github.com/GoogleContainerTools/jib/pull/2247/files#diff-aef8bf567c91df6622aa0ea29ed0c682R53

```
JIB_DEPENDECIES_TASK_NAME="jibDependencies"
```

and then reuse it in `JibPluginTest.java`.